docs/html/user_guide.rst

```diff
@@ -855,6 +855,20 @@ We are using `freeze`_ here which outputs installed packages in requirements for
 
   reqs = subprocess.check_output([sys.executable, '-m', 'pip', 'freeze'])
 
+Since pip's progress bar gets hidden when running in a subprocess, you can use
+the ``--progress-bar=json`` option for easily parsable progress information::
+
+  subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'my_package', '--progress-bar=json'])
+
+Which will give the following output after it processes each download chunk:
+
+``PROGRESS:{"current": ######, "total": ######}``
+
+Here, ``PROGRESS:`` indicates it is download progress. The rest of the message is JSON
+with the ``current`` number of bytes downloaded and ``total`` .whl size as key/value pairs.
+This can be used to build your own progress bar, or report progress in other ways.
```

Comment:
Please document that relying on the exact form of pip's output in general is unsupported, and so should not be used in production applications unless you are willing to adapt when pip's output changes.